kbase-extension/static/kbase/js/widgets/function_output/kbaseSampleSet.js

```diff
@@ -0,0 +1,228 @@
+/*
+Sebastian Le Bras - April 2020
+*/
+define ([
+    'kbwidget',
+    'bootstrap',
+    'jquery',
+    'kbase-client-api',
+    'widgets/dynamicTable',
+    'kbaseAuthenticatedWidget',
+    'kbaseTabs',
+    'kbase-generic-client-api',
+    'narrativeConfig',
+    'bluebird'
+], function(
+	KBWidget,
+	bootstrap,
+	$,
+    kbase_client_api,
+    DynamicTable,
+	kbaseAuthenticatedWidget,
+    kbaseTabs,
+    GenericClient,
+    Config,
+    Promise
+) {
+    'use strict';
+
+    return KBWidget({
+        name: 'kbaseSampleSetView',
+        parent: kbaseAuthenticatedWidget,
+        version: '1.0.0',
+        options: {
+            pageLimit: 10,
+            default_blank_value: ""
+        },
+
+        init: function init(options) {
+            this._super(options);
+
+            var self = this;
+            self.obj_ref = self.options.upas.id;
+            self.link_ref = self.obj_ref;
+
+            if(options._obj_info) {
+                self.ss_info = options._obj_info;
+                self.obj_ref = self.ss_info['ws_id'] + '/' + self.ss_info['id'] + '/' + self.ss_info['version'];
+                self.link_ref = self.ss_info['ws_id'] + '/' + self.ss_info['name'] + '/' + self.ss_info['version'];
+            }
+
+            self.client = new GenericClient(Config.url('service_wizard'), {token: self.authToken()});
+            self.ws = new Workspace(Config.url('workspace'),{'token':self.authToken()});
+
+            self.$elem.append($('<div>').attr('align', 'center').append($('<i class="fa fa-spinner fa-spin fa-2x">')));
+
+            // 1) get stats, and show the panel
+            var basicInfoCalls = [];
+            basicInfoCalls.push(
+                Promise.resolve(self.ws.get_objects2({objects: [{'ref': this.obj_ref}]}))
+                    .then(function(obj) {
+                        self.ss_obj_data = obj['data'][0]['data']
+                        self.ss_obj_info = obj['data'][0]['info']
+                        self.link_ref = self.ss_obj_info[6] + '/' + self.ss_obj_info[0] + '/' + self.ss_obj_info[4];
+                    }));
+
+            Promise.all(basicInfoCalls)
+                .then(function() {
+                    self.renderBasicTable();
+                })
+                .catch(function(err) {
+                    console.error('an error occurred! ' + err);
+                    self.$elem.empty();
+                    self.$elem.append('Error' + JSON.stringify(err));
```

Comment:
I'm not sure how helpful that error will be - if it's from the Workspace service, it might be a ginormous stack trace.